### PR TITLE
build: Update examples and base image to use Ceph v17.2.0 Quincy version

### PR DESCRIPTION
**Pull request by parth-gr**
The operator base image has been changed to
ceph v17.2.0 quincy release version `quay.io/ceph/ceph:v17.2.0`

Closes: https://github.com/rook/rook/issues/10140
Signed-off-by: parth-gr <paarora@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/cluster.yaml
Comment (subhamkrai) [inline deploy/examples/cluster.yaml:22]:
```suggestion
    # If you want to be more precise, you can always use a timestamp tag such quay.io/ceph/ceph:v17.2.0-20220420
```

Comment (parth-gr) [reply deploy/examples/cluster.yaml]: Done, where to find this timestamp? https://ceph.io/en/news/blog/2022/v17-2-0-quincy-released/

Comment (subhamkrai) [reply deploy/examples/cluster.yaml]: quay.io/ceph/ceph

**Review by leseb — CHANGES_REQUESTED** [review]
Explain in the commit message that the operator base image has now changed too.

File: deploy/examples/cluster.yaml
Comment (mykaul) [inline deploy/examples/cluster.yaml:20]: v14?

Comment (parth-gr) [reply deploy/examples/cluster.yaml]: The https://hub.docker.com/r/ceph/ceph/tags/ ul latest tag is v.16 so updated to it, will see when the v17 tag is available.

Comment (leseb) [reply deploy/examples/cluster.yaml]: The Docker Hub is not updated anymore (for a few months already), all new images are published to quay.io only.

**Review by travisn — CHANGES_REQUESTED** [review]
Let's discuss the timing on this update. In the past with previous major releases of Ceph, we haven't updated the examples and base image to use the new major release until the following minor release of Rook. This gives some time for a couple of patch releases of Ceph. In this case, this means 4.10 would be when we update to Quincy, and also remove support for Octopus.

File: Documentation/ceph-upgrade.md
Comment (BlaineEXE) [inline Documentation/ceph-upgrade.md:492]:
```suggestion
    ceph-version=v17.2.0-0
```

File: Documentation/ceph-upgrade.md
Comment (BlaineEXE) [inline Documentation/ceph-upgrade.md:494]:
```suggestion
    ceph-version=v17.2.0-0
```

File: deploy/examples/cluster.yaml
Comment (BlaineEXE) [inline deploy/examples/cluster.yaml:19]:
```suggestion
    # v15 is octopus, v16 is pacific, and v17 is quincy.
```

File: deploy/examples/cluster.yaml
Comment (BlaineEXE) [inline deploy/examples/cluster.yaml:20]:
```suggestion
    # RECOMMENDATION: In production, use a specific version tag instead of the general v17 flag, which pulls the latest release and could result in different
```

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Some nits.

File: deploy/examples/cluster.yaml
Comment (parth-gr) [reply deploy/examples/cluster.yaml]: DOne

**Review by leseb — COMMENTED** [review]
Do we need to rework your recent test that added a test that was building with quincy as a base op image?

File: images/ceph/Makefile
Comment (travisn) [inline images/ceph/Makefile:21]: I believe we want to keep the `?=` syntax so custom builds could override this if desired
```suggestion
CEPH_VERSION ?= v17.2.0-20220420
```

**Review by travisn — CHANGES_REQUESTED** [review]

**Comment by parth-gr** [comment]
> Do we need to rework your recent test that added a test that was building with quincy as a base op image?

I can create a follow-up PR, it will still work but we can change its structure.

**Review by travisn — COMMENTED** [review]
LGTM, maybe we just wait for merge until after the major doc update.

**Review by travisn — COMMENTED** [review]
@parth-gr After you fix the merge conflicts, let's go ahead and merge this, thanks.

**Comment by travisn** [comment]
@leseb Ready to approve after the merge conflict is fixed?

**Review by leseb — APPROVED** [review]

**Comment by parth-gr** [comment]
> @leseb Ready to approve after the merge conflict is fixed?

Done, thanks

**Pull request merged**
